src/main/java/com/gluonhq/strange/Step.java

```diff
@@ -55,10 +55,13 @@
 
     private boolean informal = false;
 
-    public Step() {}
+    public Step( Gate... moreGates ) {
+        this("unknown", moreGates);
+    }
 
-    public Step(String name) {
+    public Step(String name, Gate... moreGates) {
```

Comment:
the <init>(String) constructor is actually not used, so that can be removed, which will also simply this vararg constructor.
A Step is identified by its index, no need for a "name".

Reply:
Done